torch/_dynamo/utils.py

```diff
@@ -958,10 +958,13 @@ def same(
         return r
     elif is_numpy_int_type(ref) or is_numpy_float_type(ref):
         if relax_numpy_equality:
-            ref = ref.item()
+            if is_numpy_int_type(ref):
+                ref = ref.item()
+            if is_numpy_int_type(res):
+                res = res.item()
```

Comment:
Actually maybe this is wrong

Reply:
Why?

Reply:
we only tested if ref was a numpy int type, not res... actually I also see that I didn't test float. I guess I should update this.